crates/common/src/config.rs

```diff
@@ -124,7 +222,20 @@ pub struct FullNodeConfig<BitcoinServiceConfig> {
     /// Important pubkeys
     pub public_keys: RollupPublicKeys,
 }
-
+impl<BitcoinServiceConfig> FromEnv for FullNodeConfig<BitcoinServiceConfig>
+where
+    BitcoinServiceConfig: FromEnv + DeserializeOwned,
+{
+    fn from_env() -> anyhow::Result<Self> {
+        Ok(Self {
+            rpc: RpcConfig::from_env()?,
+            storage: StorageConfig::from_env()?,
+            runner: RunnerConfig::from_env().ok(),
+            da: BitcoinServiceConfig::from_env()?,
+            public_keys: RollupPublicKeys::from_env()?,
+        })
+    }
+}
```

Comment:
this can be implemented as such so that it supports all DA adapters:

```rust
impl<T FromEnv for FullNodeConfig<T>
where
    T: FromEnv + DeserializeOwned, // not sure if DeserializeOwned is needed
{
    fn from_env() -> anyhow::Result<Self> {
        Ok(Self {
            rpc: RpcConfig::from_env()?,
            storage: StorageConfig::from_env()?,
            runner: RunnerConfig::from_env().ok(),
            da: T::from_env()?,
            public_keys: RollupPublicKeys::from_env()?,
        })
    }
}
```